### PR TITLE
Feature/wf improvements

### PR DESCRIPTION
**Pull request by reyhanlioglu**
## Release Notes

- **New Features**
	- Introduced new constants for network headers: `installationId`, `instanceId`, and `workflowName`.
	- Added a new utility class `UuidUtil` for streamlined UUID generation.

- **Improvements**
	- Updated secure storage key access from `secureStorageTokenId` to `secureStorageInstallationId` across multiple components, enhancing data retrieval for analytics and device registration.
	- Streamlined UUID generation methods across various classes for consistency, including updates to the `NeoWorkflowManager`.

- **Bug Fixes**
	- Removed outdated logging related to initialization failures for NeoAdjust.

- **Chores**
	- Cleaned up import statements and removed unnecessary dependencies.

### CODE REVIEW

File: lib/core/widgets/neo_transition_listener/usecases/get_workflow_query_parameters_usecase.dart [hunk 1]
```diff
@@ -13,23 +13,24 @@
...
 import 'package:neo_core/core/storage/neo_core_secure_storage.dart';
-import 'package:uuid/uuid.dart';
+import 'package:neo_core/core/util/uuid_util.dart';
 
 class GetWorkflowQueryParametersUseCase {
   Future<String> call(NeoCoreSecureStorage secureStorage) async {
     final results = await Future.wait([
       secureStorage.read(NeoCoreParameterKey.secureStorageDeviceId),
-      secureStorage.read(NeoCoreParameterKey.secureStorageTokenId),
+      secureStorage.read(NeoCoreParameterKey.secureStorageInstallationId),
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/widgets/neo_transition_listener/usecases/get_workflow_query_parameters_usecase.dart:22]: Yes it is exist for a short time for backward compatibility

File: lib/core/analytics/neo_elastic.dart [hunk 2]
```diff
@@ -20,7 +19,7 @@ class NeoElastic {
     final packageInfo = await PackageInfo.fromPlatform();
     final results = await Future.wait([
       secureStorage.read(NeoCoreParameterKey.secureStorageDeviceId),
-      secureStorage.read(NeoCoreParameterKey.secureStorageTokenId),
+      secureStorage.read(NeoCoreParameterKey.secureStorageInstallationId),
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/analytics/neo_elastic.dart:22]: Is it intentional, tokenId will be deleted in future

File: lib/core/storage/neo_core_secure_storage.dart [hunk 3]
```diff
@@ -113,8 +112,8 @@ class NeoCoreSecureStorage {
     if (!await _storage!.containsKey(key: NeoCoreParameterKey.secureStorageDeviceInfo) && deviceInfo != null) {
       await write(key: NeoCoreParameterKey.secureStorageDeviceInfo, value: deviceInfo.encode());
     }
-    if (!await _storage!.containsKey(key: NeoCoreParameterKey.secureStorageTokenId)) {
-      await write(key: NeoCoreParameterKey.secureStorageTokenId, value: const Uuid().v1());
+    if (!await _storage!.containsKey(key: NeoCoreParameterKey.secureStorageInstallationId)) {
+      await write(key: NeoCoreParameterKey.secureStorageInstallationId, value: UuidUtil.generateUUIDWithoutHypen());
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/storage/neo_core_secure_storage.dart]: done

File: lib/core/util/uuid_util.dart [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+class UuidUtil {
+  UuidUtil._();
+
+  static String generateUUID() {
+    return const Uuid().v1();
+  }
+
+  static String generateUUIDWithoutHypen() {
+    return const Uuid().v1().replaceAll('-', '');
+  }
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/core/util/uuid_util.dart]: done

File: lib/feature/device_registration/usecases/neo_core_register_device_usecase.dart [hunk 5]
```diff
@@ -38,7 +38,7 @@ class NeoCoreRegisterDeviceUseCase {
 
       final resultArray = await Future.wait([
         secureStorage.read(NeoCoreParameterKey.secureStorageDeviceId),
-        secureStorage.read(NeoCoreParameterKey.secureStorageTokenId),
+        secureStorage.read(NeoCoreParameterKey.secureStorageInstallationId),
```
[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/feature/device_registration/usecases/neo_core_register_device_usecase.dart:41]: fixed

[the remark this answers is not in the document]
Comment (reyhanlioglu) [reply lib/feature/device_registration/usecases/neo_core_register_device_usecase.dart:41]: fixed

File: lib/core/network/models/neo_network_header_key.dart [hunk 6]
```diff
@@ -26,7 +26,10 @@ abstract class NeoNetworkHeaderKey {
   static const deviceModel = "X-Device-Model";
   static const devicePlatform = "X-Device-Platform";
   static const deviceVersion = "X-Device-Version";
+  static const installationId = "X-Installation-Id";
```
Comment (alperen-commencis) [inline lib/core/network/models/neo_network_header_key.dart:29]: analizde tüm harfler küçük görünüyor bu parametreler için, sıkıntı yok heralde böyle

**Review by alperen-commencis — APPROVED** [review]

File: lib/core/network/models/neo_network_header_key.dart (hunk 6 above)
Comment (reyhanlioglu) [reply lib/core/network/models/neo_network_header_key.dart:29]: onlar küçültüyo sanırım ama doğru format bu

**Pull request merged**